app/lib/features/chat/widgets/text_message_builder.dart

```diff
@@ -52,7 +52,7 @@ class _TextMessageBuilderConsumerState
     //remove mx-reply tags.
     String parsedString = simplifyBody(widget.message.text);
     final urlRegexp = RegExp(
-      r'https://matrix\.to/#/@[A-Za-z0-9]+:[A-Za-z0-9]+\.[A-Za-z0-9]+',
+      r'https://matrix\.to/#/@[A-Za-z0-9\-]+:[A-Za-z0-9\-]+\.[A-Za-z0-9\-]+',
```

Comment:
this screams for having tests for ...

Reply:
Tested the above Regex with a couple of usernames and it works fine. Still, you can guide me in case of any missing or incorrect part.